src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -624,6 +651,10 @@ export class EditorPart extends Part implements IEditorPart {
 
 	private doSetGroupActive(group: IEditorGroupView): void {
 		if (this._activeGroup !== group) {
+			if (this._activeGroup && this.isGroupMaximized(this._activeGroup)) {
```

Comment:
I actually think we have to have these 3 lines in the `doRestoreGroup` method below. That method is being called in some other cases (from `editorGroupView`): it can happen that a group should restore without activating it. I would make it that `unmaximizeGroup` is called when the group to restore is minimized.